packages/core-ethereum/crates/core-ethereum-rpc/src/indexer.rs

```diff
@@ -0,0 +1,360 @@
+                                current_block_log.block_id = log.block_number;
+                            }
+
+                            debug!("retrieved {log}");
+                            current_block_log.logs.push(log);
+                        }
+
+                        debug!("retrieved complete {current_block_log}");
+                        yield current_block_log;
+                    }
+                    Err(e) => error!("failed to obtain current block number from chain: {e}")
+                }
+
+                sleep(self.cfg.expected_block_time).await;
+            }
+        }))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::time::Duration;
+
+    use ethers::contract::EthEvent;
+    use ethers_providers::Middleware;
+    use futures::StreamExt;
+
+    use bindings::hopr_channels::*;
+    use bindings::hopr_token::{ApprovalFilter, HoprToken, TransferFilter};
+    use core_crypto::keypairs::{ChainKeypair, Keypair};
+    use core_ethereum_types::{create_anvil, create_rpc_client_to_anvil, ContractAddresses, ContractInstances};
+    use utils_log::debug;
+    use utils_types::primitives::Address;
+
+    use crate::client::tests::ReqwestRequestor;
+    use crate::client::{JsonRpcProviderClient, SimpleJsonRpcRetryPolicy};
+    use crate::rpc::tests::mint_tokens;
+    use crate::rpc::{RpcOperations, RpcOperationsConfig};
+    use crate::{BlockWithLogs, HoprIndexerRpcOperations, LogFilter};
+
+    async fn fund_channel<M: Middleware + 'static>(
+        counterparty: Address,
+        hopr_token: HoprToken<M>,
+        hopr_channels: HoprChannels<M>,
+    ) {
+        hopr_token
+            .approve(hopr_channels.address(), 1u128.into())
+            .send()
+            .await
+            .unwrap()
+            .await
+            .unwrap();
+
+        hopr_channels
+            .fund_channel(counterparty.into(), 1u128)
+            .send()
+            .await
+            .unwrap()
+            .await
+            .unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_should_get_block_number() {
+        let anvil = create_anvil(Some(Duration::from_secs(1)));
+        let chain_key_0 = ChainKeypair::from_secret(anvil.keys()[0].to_bytes().as_ref()).unwrap();
+
+        let client = JsonRpcProviderClient::new(&anvil.endpoint(), ReqwestRequestor::default());
+
+        let rpc = RpcOperations::new(client, &chain_key_0, Default::default(), SimpleJsonRpcRetryPolicy)
+            .expect("failed to construct rpc");
+
+        let b1 = rpc.block_number().await.expect("should get block number");
+        tokio::time::sleep(Duration::from_secs(2)).await;
+        let b2 = rpc.block_number().await.expect("should get block number");
+
+        assert!(b2 > b1, "block number should increase");
+    }
+
+    #[tokio::test]
+    async fn test_try_stream_logs_should_contain_all_logs_when_opening_channel() {
+        let _ = env_logger::builder().is_test(true).try_init();
```

Comment:
Remnant?

Reply:
Why not keep it

Reply:
Log spam in the tests.

Reply:
Only if we pass `RUST_LOG=info` or more in env, tests have `RUST_LOG=error`